pkg/cmd/attestation/trustedroot/trustedroot.go

```diff
@@ -93,37 +124,53 @@ func getTrustedRoot(makeTUF tufClientInstantiator, opts *Options) error {
 
 		tufOpt.Root = tufRoot
 		tufOpt.RepositoryBaseURL = opts.TufUrl
-		tufOptions = append(tufOptions, tufOpt)
+		tufOptions = append(tufOptions, tufConfig{
+			tufOptions: tufOpt,
+			targets:    []string{defaultTR},
+		})
 	} else {
 		// Get from both Sigstore public good and GitHub private instance
-		tufOptions = append(tufOptions, tufOpt)
+		tufOptions = append(tufOptions, tufConfig{
+			tufOptions: tufOpt,
+			targets:    []string{defaultTR},
+		})
 
 		tufOpt = verification.GitHubTUFOptions()
 		tufOpt.CacheValidity = 0
-		tufOptions = append(tufOptions, tufOpt)
+		targets := []string{defaultTR}
+		if opts.TrustDomain != "" {
+			targets = append(targets, fmt.Sprintf("%s.%s",
+				opts.TrustDomain, defaultTR))
+		}
+		tufOptions = append(tufOptions, tufConfig{
+			tufOptions: tufOpt,
+			targets:    targets,
+		})
 	}
 
-	for _, tufOpt = range tufOptions {
-		tufClient, err := makeTUF(tufOpt)
+	for _, tufOpt := range tufOptions {
+		tufClient, err := makeTUF(tufOpt.tufOptions)
 		if err != nil {
 			return fmt.Errorf("failed to create TUF client: %v", err)
 		}
 
-		t, err := tufClient.GetTarget("trusted_root.json")
-		if err != nil {
-			return err
-		}
+		for _, target := range tufOpt.targets {
+			t, err := tufClient.GetTarget(target)
+			if err != nil {
+				return err
```

Comment:
If we can't reach the target here, should the client facing error message include some content about checking the URL itself? Maybe the returned error already includes something like this.

Reply:
The error will contain context concerning the target, but we can do better. Let me add more context to the error message.